src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -1031,7 +1034,34 @@ public List<Setting<?>> getSettings() {
             settings.add(Setting.simpleString(ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, Property.NodeScope, Property.Filtered));
             settings.add(Setting.simpleString(ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_ERROR_MESSAGE, Property.NodeScope, Property.Filtered));
 
-            
+            settings.add(
+                    Setting.intSetting(
+                            ConfigConstants.SECURITY_RESTAPI_PASSWORD_MIN_LENGTH,
+                            -1, -1, Property.NodeScope, Property.Filtered));
+            settings.add(Setting.simpleString(
+                    ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_STRENGTH,
+                    PasswordValidator.ScoreStrength.STRONG.name().toLowerCase(Locale.ROOT),
+                    value -> {
+                        if (!Strings.isNullOrEmpty(value)) {
+                            final String errorMessage =
+                                    "Setting ["
+                                            + ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_STRENGTH
+                                            + "] cannot be used with the configured: " + value
+                                            + ". Expected one of ["
+                                            + PasswordValidator.ScoreStrength.expectedValues()
+                                            + "]";
+                            try {
+                                final PasswordValidator.ScoreStrength scoreStrength =
+                                        PasswordValidator.ScoreStrength.valueOf(value.toUpperCase(Locale.ROOT));
+                                if (scoreStrength == PasswordValidator.ScoreStrength.WEAK) {
```

Comment:
If weak isn't something that we'd want to approve, why not remove it from the enum to keep this error handling simplier?

Reply:
Then `expectedValues()` can be removed and replaced with  string joining collector on the ScoreStrength values.

Reply:
Sure forgaet about `StringJoiner` :-)

Reply:
> If weak isn't something that we'd want to approve, why not remove it from the enum to keep this error handling simplier?

I think yes.

Reply:
Left it since library returns score equals to 1 and added a comment why i was defined.